tracer/src/Datadog.Trace/Activity/OtlpHelpers.cs

```diff
@@ -0,0 +1,383 @@
+
+                // Fallback: Use the information provided by Activity
+                if (activity5 is not null)
+                {
+                    span.ResourceName = activity5.DisplayName;
+                }
+                else
+                {
+                    span.ResourceName = activity.OperationName;
+                }
+            }
+
+            // Update Type with a reasonable default
+            if (string.IsNullOrWhiteSpace(span.Type))
+            {
+                span.Type = activity5 is null ? SpanTypes.Custom : AgentSpanKind2Type(activity5.Kind, span);
+            }
+        }
+
+        internal static void SetTagObject(Span span, string key, object? value)
+        {
+            if (value is null)
+            {
+                span.SetTag(key, null);
+                return;
+            }
+
+            switch (value)
+            {
+                case char c: // TODO: Can't get here from OTEL API, test with Activity API
+                    AgentSetOtlpTag(span, key, c.ToString());
+                    break;
+                case string s:
+                    AgentSetOtlpTag(span, key, s);
+                    break;
+                case bool b:
+                    AgentSetOtlpTag(span, key, b ? "true" : "false");
+                    break;
+                case byte b: // TODO: Can't get here from OTEL API, test with Activity API
+                    span.SetMetric(key, b);
+                    break;
+                case sbyte sb: // TODO: Can't get here from OTEL API, test with Activity API
+                    span.SetMetric(key, sb);
+                    break;
+                case short sh: // TODO: Can't get here from OTEL API, test with Activity API
+                    span.SetMetric(key, sh);
+                    break;
+                case ushort us: // TODO: Can't get here from OTEL API, test with Activity API
+                    span.SetMetric(key, us);
+                    break;
+                case int i: // TODO: Can't get here from OTEL API, test with Activity API
+                    span.SetMetric(key, i);
+                    break;
+                case uint ui: // TODO: Can't get here from OTEL API, test with Activity API
+                    span.SetMetric(key, ui);
+                    break;
+                case long l: // TODO: Can't get here from OTEL API, test with Activity API
+                    span.SetMetric(key, l);
+                    break;
+                case ulong ul: // TODO: Can't get here from OTEL API, test with Activity API
+                    span.SetMetric(key, ul);
+                    break;
+                case float f: // TODO: Can't get here from OTEL API, test with Activity API
+                    span.SetMetric(key, f);
+                    break;
+                case double d:
+                    span.SetMetric(key, d);
+                    break;
+                case Array array:
```

Comment:
Should make this `IEnumerable` to allow handling more collection types?

Reply:
Makes sense to me. Since the OTEL API only allows you to set values to primitives or arrays of primitives, I won't be able to test `List<T>` in this PR, but we can do that in the Activity PR that Steven is working on

Reply:
Done in c2fdbdc113a6a7634f1c6b313c38f39e85374fdf

Reply:
Thanks